### ISSUE TITLE
Change Request: Ignore linguist-generated files by default

### ISSUE THREAD

**Issue by matthieu-crouzet** [eslint/eslint#20730]
### ESLint version

v9.39.4

### What problem do you want to solve?

In my project, we have some files marked as `linguist-generated` in the `.gitattributes`.
The issue is that we need to specify them in the `globalIgnores`

### What do you think is the correct solution?

IMO it doesn't make sense to lint generated files, so I would expect these files to be ignored as if they are in the `.gitignore`.
We could imagine a method as `includeIgnoreFile` if someone need to include them.

### Participation

- [ ] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
Currently ESLint does not have any git integration, it just supports the same glob patterns as the `.gitignore` file.
The feature you are describing could be accomplished in userland by reading the `.gitattributes` file explicitly and converting it to a global ignore (similar to the existing [`includeIgnoreFile`](https://eslint.org/docs/latest/use/configure/ignore#include-gitignore-files)):
```javascript
import { ignoreFilesWithGitAttribute } from "helper";

export default [
	ignoreFilesWithGitAttribute("./.gitattributes", ["linguist-generated"]),
];
```
I do not think this should be default ESLint functionality but I can see the use case for providing this helper.

**Comment by kirkwaiblinger** [comment]
Related to https://github.com/eslint/rewrite/issues/329 (not a duplicate or anything just linking for visibility)

**Comment by JoshuaKGoldberg** [comment]
+1 to providing a helper for this. Additional +1 to making it mirror `includeIgnoreFile`.

**Comment by JoshuaKGoldberg** [comment]
I think this is not stale, just waiting for more input from @eslint/eslint-team.

**Comment by JoshuaKGoldberg** [comment]
I think this is not stale, just waiting for more input from @eslint/eslint-team.

**Comment by DMartens** [comment]
I hope this is not stale, we just need more input from @eslint/eslint-team.

**Comment by lumirlumir** [comment]
**TSC Summary**: This issue proposes ignoring files marked with attributes such as `linguist-generated` in `.gitattributes`. The discussion so far suggests that this should not be default ESLint behavior, but there is support for providing a helper similar to `includeIgnoreFile()`.

**TSC Question**: Should ESLint provide an official helper for ignoring files based on attributes specified in `.gitattributes`, or should this remain a userland concern?

**Comment by sethamus** [comment]
Now that this has been accepted, I'll work on it unless someone else on the team wants to pick it up.
